BH TTNN Unit failure in test for uniform

https://github.com/tenstorrent/tt-metal/actions/runs/14417681705/job/40436423449
```
FAILED tests/ttnn/unit_tests/operations/test_uniform.py::test_uniform[seed=2024-dtype=bfloat16-rand_range=[0, 1]-shape=[32, 32]] - RuntimeError: TT_THROW @ /work/tt_metal/impl/program/program.cpp:113: tt::exception
info:
Failed to generate binaries for compute_uniform TT_THROW @ /work/tt_metal/jit_build/build.cpp:67: tt::exception
info:
trisc1 build failed. Log: In file included from /work/ttnn/cpp/ttnn/operations/uniform/device/kernels/compute_uniform.cpp:7,
                 from ../chlkc_math.cpp:3,
                 from /work/.ttnn_runtime_artifacts/tt_metal/hw/ckernels/blackhole/metal/common/chlkc_list.h:21,
                 from /work/.ttnn_runtime_artifacts/tt_metal/hw/firmware/src/trisck.cc:14:
/work/.ttnn_runtime_artifacts/tt_metal/include/compute_kernel_api/eltwise_unary/rand.h:9:10: fatal error: llk_math_eltwise_unary_sfpu_rand.h: No such file or directory
    9 | #include "llk_math_eltwise_unary_sfpu_rand.h"
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
```
I think this is similar to other eltwise ops.

Needs to be fixed.

--- a/tests/ttnn/unit_tests/operations/test_uniform.py
+++ b/tests/ttnn/unit_tests/operations/test_uniform.py
@@ -15,7 +15,6 @@ from tests.ttnn.unit_tests.operations.test_utils import (
     compute_kernel_ids,
     get_lib_dtype,
 )
-from models.utility_functions import skip_for_grayskull
 from enum import Enum
 
 
@@ -96,7 +95,6 @@ def run_uniform(shape, rand_range, dtype, device, seed=0, compute_kernel_options
         )
 
 
-@skip_for_grayskull("Requires wormhole_b0 to run")
 @pytest.mark.parametrize(
     "shape",
     [
@@ -115,7 +113,6 @@ def test_uniform(shape, rand_range, dtype, seed, device):
     run_uniform(shape, rand_range, dtype, device, seed=seed)
 
 
-@skip_for_grayskull("Requires wormhole_b0 to run")
 @pytest.mark.parametrize(
     "shape",
     [[2, 32, 32, 16]],
@@ -140,7 +137,6 @@ def test_uniform_callback(shape, rand_range, dtype, seed, device, use_program_ca
     assert num_program_cache_entries_list[0] == num_program_cache_entries_list[1]
 
 
-@skip_for_grayskull("Requires wormhole_b0 to run")
 @pytest.mark.parametrize(
     "shape",
     [[512, 512], [5, 2, 4, 70, 40]],

--- a/tt_metal/hw/ckernels/blackhole/metal/llk_api/llk_sfpu/ckernel_sfpu_rand.h
+++ b/tt_metal/hw/ckernels/blackhole/metal/llk_api/llk_sfpu/ckernel_sfpu_rand.h
@@ -39,14 +39,12 @@ inline void rand(uint32_t from, uint32_t scale) {
         // -1 to ensure the float is within the range [0, 1).
         // lreg0 = lreg0 - 1
         TTI_SFPADDI(0xbf80 /*-1*/, p_sfpu::LREG0, 0);
-        TTI_SFPNOP;
 
         // Scale the float from [0, 1) to [from, from + scale)
         // lreg0 = lreg0 * scale + from
         TTI_SFPMAD(p_sfpu::LREG0, p_sfpu::LREG1, p_sfpu::LREG2, p_sfpu::LREG0, 0);
-        TTI_SFPNOP;
 
-        TTI_SFPSTORE(0, 3, 7, 0);
+        TTI_SFPSTORE(p_sfpu::LREG0, p_sfpu::LREG3, ADDR_MOD_7, 0);
         dst_reg++;
     }
 }

--- a/tt_metal/hw/ckernels/blackhole/metal/llk_api/llk_sfpu/ckernel_sfpu_rand.h
+++ b/tt_metal/hw/ckernels/blackhole/metal/llk_api/llk_sfpu/ckernel_sfpu_rand.h
@@ -1,0 +1,53 @@
+// SPDX-FileCopyrightText: © 2025 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+#pragma once
+
+#include "ckernel.h"
+#include "ckernel_defs.h"
+
+using namespace sfpi;
+
+namespace ckernel::sfpu {
+
+template <bool APPROXIMATION_MODE>
+inline void rand_init(uint32_t seed) {
+    init_prng_seed(seed);
+}
+
+template <bool APPROXIMATION_MODE>
+inline void rand(uint32_t from, uint32_t scale) {
+    // Load scale param to lreg1
+    TT_SFPLOADI(p_sfpu::LREG1, 10, scale & 0xFFFF);
+    TT_SFPLOADI(p_sfpu::LREG1, 8, scale >> 16);
+
+    // Load from param to lreg2
+    TT_SFPLOADI(p_sfpu::LREG2, 10, from & 0xFFFF);
+    TT_SFPLOADI(p_sfpu::LREG2, 8, from >> 16);
+
+#pragma GCC unroll 0
+    for (int d = 0; d < 8; d++) {
+        // Generate random float
+        TTI_SFPMOV(0, 9, p_sfpu::LREG0, 8);
+
+        // Unset sign bit and Set exponent to 127 to ensure the float is within the range [1, 2).
+        // lreg0.sign = 0
+        // lreg0 = {sign: 0, exponent: 127, mantissa: lreg0.mantissa}
+        TTI_SFPSETSGN(0, p_sfpu::LREG0, p_sfpu::LREG0, 1);
+        TTI_SFPSETEXP(127, p_sfpu::LREG0, p_sfpu::LREG0, 1);
+
+        // -1 to ensure the float is within the range [0, 1).
+        // lreg0 = lreg0 - 1
+        TTI_SFPADDI(0xbf80 /*-1*/, p_sfpu::LREG0, 0);
+        TTI_SFPNOP;
+
+        // Scale the float from [0, 1) to [from, from + scale)
+        // lreg0 = lreg0 * scale + from
+        TTI_SFPMAD(p_sfpu::LREG0, p_sfpu::LREG1, p_sfpu::LREG2, p_sfpu::LREG0, 0);
+        TTI_SFPNOP;
+
+        TTI_SFPSTORE(0, 3, 7, 0);
+        dst_reg++;
+    }
+}
+}  // namespace ckernel::sfpu

--- a/tt_metal/hw/ckernels/blackhole/metal/llk_api/llk_sfpu/ckernel_sfpu_rand.h
+++ b/tt_metal/hw/ckernels/blackhole/metal/llk_api/llk_sfpu/ckernel_sfpu_rand.h
@@ -44,7 +44,7 @@ inline void rand(uint32_t from, uint32_t scale) {
         // lreg0 = lreg0 * scale + from
         TTI_SFPMAD(p_sfpu::LREG0, p_sfpu::LREG1, p_sfpu::LREG2, p_sfpu::LREG0, 0);
 
-        TTI_SFPSTORE(p_sfpu::LREG0, p_sfpu::LREG3, ADDR_MOD_7, 0);
+        TTI_SFPSTORE(p_sfpu::LREG0, FP32, ADDR_MOD_7, 0);
         dst_reg++;
     }
 }

--- a/tt_metal/hw/ckernels/blackhole/metal/llk_api/llk_sfpu/llk_math_eltwise_unary_sfpu_rand.h
+++ b/tt_metal/hw/ckernels/blackhole/metal/llk_api/llk_sfpu/llk_math_eltwise_unary_sfpu_rand.h
@@ -1,0 +1,27 @@
+// SPDX-FileCopyrightText: © 2025 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include "ckernel_instr_params.h"
+#include "ckernel_sfpu_rand.h"
+#include "llk_math_eltwise_unary_sfpu_init.h"
+#include "llk_math_eltwise_unary_sfpu_params.h"
+
+namespace ckernel {
+
+// New LLK SFPU APIs
+
+template <bool APPROXIMATE>
+inline void llk_math_eltwise_unary_sfpu_rand_init(uint32_t seed = 0) {
+    llk_math_eltwise_unary_sfpu_init<SfpuType::unused, APPROXIMATE>(sfpu::rand_init<APPROXIMATE>, seed);
+}
+
+template <bool APPROXIMATE>
+inline void llk_math_eltwise_unary_sfpu_rand(uint32_t dst_index, uint32_t from, uint32_t scale) {
+    llk_math_eltwise_unary_sfpu_params<APPROXIMATE>(
+        ckernel::sfpu::rand<APPROXIMATE>, dst_index, VectorMode::RC, from, scale);
+}
+
+}  // namespace ckernel

